### PR TITLE
Development

### PR DESCRIPTION
**Pull request by davitbadikyan**
Please review the code for further updates

### CODE REVIEW

File: src/com/company/Service.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    private double inUniOneSubjectAvarage;
+    private int inFacOneSubjectAllGrade = 0;
+    private int inFacOneSubjectStudentsCount = 0;
+    private double inFacOneSubjectAvarage;
+    private int inGroupOneSubjectAllGrade = 0;
+    private int inGroupOneSubjectStudentsCount = 0;
+    private double inGroupOneSubjectAvarage;
+    private double studentGpa;
+
+    public void UniversityStructure() {
```
Comment (ando7555) [inline src/com/company/Service.java]: Write method name with java conventions, separate creation logic from GPA counting.

**Review by ando7555 — COMMENTED** [review]
Place of saving group count you can check the length of group array and that time throw a runtime exception,
same for other classes.
Throw runtime exception place of custom exceptions.

**Review by ando7555 — COMMENTED** [review]
Remove redundant things from  .gitgnore
You do not need to save uinversityName, groupName, facultyName inside Student class, there are already exist inside their appropriate classes.

**Review by ando7555 — COMMENTED** [review]
It would be better to throw a custom exception in the process of calculating GPA and  do null check  to avoid NPE

File: src/com/university_project/Service.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+                    inUniOneSubjectStudentsCount++;
+                }
+            }
+        }
+
+        this.inUniOneSubjectAverage = inUniOneSubjectAllGrade / inUniOneSubjectStudentsCount;
+        System.out.println("University's average grade from Math is " + this.inUniOneSubjectAverage);
+    }
+
+    public void facultyFromOneSubjectGPA() {
```
Comment (ando7555) [inline src/com/university_project/Service.java]: Here your method must accept university and subject parameters. It is not good when you immediately compare values. For example, you have done that thing with "Physics" Do the same for other methods.

**Review by avetisyanAnna — APPROVED** [review]

**Review by ando7555 — APPROVED** [review]

**Pull request merged**